### ISSUE TITLE
CJSON: Patch for cjson in stable tarantool, admitting encode/decode cdata number

### ISSUE THREAD

**Issue by orchaton** [tarantool/tarantool#1689]
Added some function from version 1.6, allowing encoding and decoding to json cdata type (int64 and uint64). Could you add it to project? :)
 
[format-patch.zip](https://github.com/tarantool/tarantool/files/413859/patch.zip)

**Comment by rtsisyk** [comment]
```
 error: ‘json_append_cdata’ defined but not used [-Werror=unused-function]
 static void json_append_cdata(lua_State *l, json_config_t *cfg,
```

I added cjson files to the tarantool source tree and removed submodule. Could you please re-base your patch on gh-1689-cjson-cdata-1.5 branch?

**Comment by orchaton** [comment]
Of course. Here: 
[format-patch.zip](https://github.com/tarantool/tarantool/files/418537/format-patch.zip)

That error was a mistake. Now it's been fixed.

**Comment by rtsisyk** [comment]
@orchaton patch is OK, thanks a lot! I ported latest version of pushcdata/checkcdata/pushint64/pushuint64/ and other cdata functions to src/lua/utils.c and rebased your patch. See gh-1689-cjson-cdata-1.5-v2 branch.

Last but not least thing: please add minimum viable test cases for int64/uint64 encoding/decoding to test/box/cjson.test. 

https://github.com/tarantool/tarantool/blob/1.7/test/app-tap/lua/serializer_test.lua#L27

**Comment by orchaton** [comment]
Updating tests using "old" style. Path: test/box/cjson.test
[format-patch.tests.zip](https://github.com/tarantool/tarantool/files/422939/format-patch.tests.zip)

**Comment by rtsisyk** [comment]
Fixed in 1.5.5-33-g38b2398
